### PR TITLE
[gptq] faster gptq cuda kernel

### PR DESCRIPTION
**Pull request by Xu-Kai**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Comment by Xu-Kai** [comment]
image

File: colossalai/gptq/cai_gptq/cai_quant_linear.py [hunk 1]
```diff
@@ -101,21 +122,75 @@ def pack(self, linear, scales, zeros, g_idx=None):
...
             self.g_idx = None
         else:
             self.g_idx = g_idx
 
+        CaiQuantLinear.max_dq_buffer_size = max(CaiQuantLinear.max_dq_buffer_size, self.qweight.numel() * 8)
 
-    def forward(self, x):
+        if self.g_idx is not None:
+            CaiQuantLinear.max_inner_outer_dim = max(CaiQuantLinear.max_inner_outer_dim, self.infeatures, self.outfeatures)
+            max_input_len=4096
```
Comment (ver217) [inline colossalai/gptq/cai_gptq/cai_quant_linear.py:134]: Is this constant related to kernels?

Comment (Xu-Kai) [reply colossalai/gptq/cai_gptq/cai_quant_linear.py:134]: Yes. I don't know how to set properly.

**Review by ver217 — APPROVED** [review]

**Comment by Xu-Kai** [comment]
image
